contents/handbook/people/team-structure/product-analytics/objectives.md

```diff
@@ -0,0 +1,91 @@
+> - Objective: Implement most frequently requested features with 3+ happy companies for the feature launch
+>   - Key results:
+>     - Implement the following:
+>       - Multiple breakdowns
+>       - Cohort style queries (did x event x times), but within a query itself
+>       - Composite events: ‘Any event’, ‘New user’ (users first event), ‘top events’
+>       - User composition (ie pie chart where you can select properties)/
+>       - Superscale querying - Get 98% of each company’s insights P99 below an acceptable time (to be worked out by the team)
+> - Why?
+>   - Quality/parity with other product analytics tools. Delighting customers especially power users, will lead to word of mouth growth.
+
+
+
+## Proposed Objective 1: Nail Data Exploration
+
+NB! This objective still needs analysis and clarification
+
+### Why?
+
+Product reason:
+
+- We have a pretty good interface for data *visualization* (trends, funnels, retention, etc), but not so much for data 
+  *exploration* (events, persons). Users have been asking for more.
+- The biggest pain comes from not being able to move between different types of related data. Imagine [this scenario](https://github.com/PostHog/posthog/issues/7963#issuecomment-1010235411) (© @rcmarron)
+> 1) "I wonder which organizations have used feature Y"
+> 2) Search for that event/action and change the tab to organizations
+> 3) "Interesting CompanyXYZ is using it, I wonder who"
+> 4) Flip to persons + add an organization filter for CompanyXYZ
+> 5) See the persons
+> 6) "I wonder how/why they're using it"
+> 7) Click recordings + see the recordings
+> It would be pretty clunky/impossible to explore your data in that way with today's interface. 
+- Currently you need to re-create this same filtered query on 3-4 different pages and manually copy over the properties. 
+  It's doable, but not a great experience. 
+
+Code reason:
+- We have a lot of bespoke bridges to show related data in the interface (people on insights, recordings on people).
+- Every new one adds more complexity. 
+- This compounded by ever so slightly different use of filters on different data types.
+- The further we continue down this path, the harder it'll be to have a fully integrated product across all data.
+
+Meta reason:
+- We now have separate product teams. 
+- If we're not careful, we'll end up with inoperable products. 
+- This builds the first shared bridge between all of them.
+
+### How?
+- Universal Search: https://github.com/PostHog/posthog/issues/7963
+- An "explore" interface/table, which supports input from various swappable data sources, with a unified set of filters applied to data.
+- Support for custom columns and [formulas](https://github.com/PostHog/posthog/issues/11913) on those tables.
+- Ability to save as a cohorts, save as an action, or export as a CSV when the data permits.
+- The ability to link to these tables from anywhere. "Explore further" links wherever we show *any* data in the interface.
```

Comment:
I like the idea of a powerful unified querying interface which means that improvements to the insights part also improve the session recording part etc.

I also like the ability to quickly pivot between session recordings, cohorts, orgs, people, insights - (here's another example where this could be powerful https://github.com/PostHog/posthog/issues/11942)

The bit I get worried about is the natural language interface for this - this seems separate and less useful for me as a user.

Reply:
> The question I'd have is would these move the needle enough for the experience as well as the metrics that we care about?

Genuine question, what are the metrics we care about? 

> The bit where I get more worried is then including a natural language interface for this - this seems separate and less useful for me as a user. In google analytics I never used the sentence querying for example

I'm very happy to scrap the free form textfield part of this, and replace it with a select field. The screenshot is meant to be aspirational. I pointed out as well that this can be a follow up, if anything.

Reply:
Note that the ticket wasn't intended to be about free-form text search - the section that might be confusing is more about how the structure of the search might look like.

Updated the ticket https://github.com/PostHog/posthog/issues/7963

> Similar to insights, this would take the form of a sentence-like structure containing dropdowns, like:
>
> Show me [People | Group Type G | Cohorts | Recordings | Events from people | Recordings from people] who [Have property | Belong to COHORT | Have done Y]